protocol/synthetix/contracts/storage/DelegationIntent.sol

```diff
@@ -0,0 +1,165 @@
+         */
+        uint128 accountId;
+        /**
+         * @notice The ID of the pool for which the account has an outstanding intent to delegate a new amount of collateral to
+         */
+        uint128 poolId;
+        /**
+         * @notice The address of the collateral type that the account has an outstanding intent to delegate a new amount of
+         */
+        address collateralType;
+        /**
+         * @notice The delta amount of collateral that the account has an
+         * outstanding intent to delegate/undelegate to the pool,
+         * denominated with 18 decimals of precision
+         */
+        int256 deltaCollateralAmountD18;
+        /**
+         * @notice The intended amount of leverage associated with the new
+         * amount of collateral that the account has an outstanding intent
+         * to delegate to the pool
+         * @dev The system currently only supports 1x leverage
+         */
+        uint256 leverage;
+        /**
+         * @notice The timestamp at which the intent was declared
+         */
+        uint32 declarationTime;
+    }
+
+    /**
+     * @dev Returns the delegation intent stored at the specified nonce id.
+     */
+    function load(uint256 id) internal pure returns (Data storage delegationIntent) {
+        bytes32 s = keccak256(abi.encode("io.synthetix.synthetix.DelegationIntent", id));
+        assembly {
+            delegationIntent.slot := s
+        }
+    }
+
+    /**
+     * @dev Returns the delegation intent stored at the specified nonce id. Checks if it's valid
+     */
+    function loadValid(uint256 id) internal view returns (Data storage delegationIntent) {
+        delegationIntent = load(id);
+
+        if (delegationIntent.id != id) {
+            revert InvalidDelegationIntentId();
+        }
+    }
+
+    function latestId() internal view returns (uint256) {
+        return Config.readUint(_ATOMIC_VALUE_LATEST_ID, 0);
+    }
+
+    function nextId() internal returns (uint256 id) {
+        id = Config.readUint(_ATOMIC_VALUE_LATEST_ID, 0) + 1;
+        Config.put(_ATOMIC_VALUE_LATEST_ID, bytes32(id));
+    }
+
+    function processingStartTime(Data storage self) internal view returns (uint32) {
+        (uint32 requiredDelayTime, ) = Pool
+            .loadExisting(self.poolId)
+            .getRequiredDelegationDelayAndWindow(self.deltaCollateralAmountD18 < 0);
+        return self.declarationTime + requiredDelayTime;
+    }
+
+    function processingEndTime(Data storage self) internal view returns (uint32) {
+        (uint32 requiredDelayTime, uint32 requiredWindowTime) = Pool
+            .loadExisting(self.poolId)
+            .getRequiredDelegationDelayAndWindow(self.deltaCollateralAmountD18 < 0);
+
+        // Apply default (forever) window time if not set
+        if (requiredWindowTime == 0) {
+            requiredWindowTime = 86400 * 360; // 1 year
```

Comment:
I understand we discussed this before, and sorry to bring it up again, but in case a window is shortened, would it be possible to use the min between the shortened window and the one saved when the intent was specified?
Like my thoughts is that we raise the window, or misconfigure to 0, people now have to wait a year to expire. so we need owner to kil those events that can't be executed (due to issue)  nor expired... So we shorten the window... 
I understand the double edge sword... I'll think about this some more, but want to avoid the above scenario of peole having to wait a long time for removing / killing a non-executable intent.